src/sst/core/warnmacros.h

```diff
@@ -14,7 +14,7 @@
 #ifndef SST_CORE_WARNMACROS_H
 #define SST_CORE_WARNMACROS_H
 
-#define UNUSED(x) x __attribute__((unused))
+#define UNUSED(x) x [[maybe_unused]]
```

Comment:
I haven't checked with the arm compiler, but maybe `#define UNUSED(x) [[maybe_unused]] x`?

Reply:
> I haven't checked with the arm compiler, but maybe `#define UNUSED(x) [[maybe_unused]] x`?

I think that fails compilation with some current usages of `UNUSED()`. The macro calls would need to be changed. See my previous commend on the combinations of `[[maybe_unused]]` and their relationships to how the macro is called.

The current PR on Core builds on almost all platforms, because it puts `[[maybe_unused]]` after the macro argument, which is usually a variable name without a type. If we use your suggestion, all macro calls would need to be changed to have the complete type and variable passed to the macro, like `UNUSED(int x[10])`, while it now uses `int UNUSED(x[10])` or `int UNUSED(x)[10]`.